File: csharp/kindergarten-garden/KindergartenGarden.cs
using System;
using System.Collections.Generic;

public enum Plant
{
    Violets,
    Radishes,
    Clover,
    Grass
}

public class KindergartenGarden
{    
    private string[] garden;
    private static Dictionary<char, Plant> map = new Dictionary<char, Plant>(){
        { 'V', Plant.Violets },
        { 'R', Plant.Radishes },
        { 'C', Plant.Clover },
        { 'G', Plant.Grass },
    };
    public KindergartenGarden(string diagram)
    {
        garden = diagram.Split(new []{ '\n' });
    }

    public IEnumerable<Plant> Plants(string student)
    {
        return new Plant[]{
            map[garden[0][(student[0] - 'A') * 2]],
            map[garden[0][((student[0] - 'A') * 2) + 1]],
            map[garden[1][(student[0] - 'A') * 2]],
            map[garden[1][((student[0] - 'A') * 2) + 1]],
        };
    }
}

File: kotlin/kindergarten-garden/src/main/kotlin/KindergartenGarden.kt
var GardenMap = mapOf('C' to "clover", 'G' to "grass", 'R' to "radishes", 'V' to "violets")

val nucleotideCounts = mutableMapOf('A' to 0, 'C' to 0, 'G' to 0, 'T' to 0)

class KindergartenGarden(private val diagram: String) {

    fun getPlantsOfStudent(student: String): List<String> {

        var index = (student.get(0).toUpperCase().toInt() - 'A'.toInt()) * 2

        return diagram.split('\n').fold(listOf<String>()) { accum, row -> accum + 
            GardenMap.getOrDefault(row[index], "") + GardenMap.getOrDefault(row[index + 1], "")
        }
    }
}
